### ISSUE TITLE
`load_all()` use `file_format`

### ISSUE THREAD

**Issue by Ethan-DeBandi99** [Bears-R-Us/arkouda#1324]
Most of the save code has been updated to use `file_format` to indicate if saving should be done via Parquet or HDF5 (defaulting to HDF5). Working on #1303, I noticed that `pdarrayIO:load_all()` is not using this. But instead uses `os.path.splitext(path_prefix)` to access the filetype and format the filepath for configuring the dataset. The issue here is that a user could save to a directory with no name appended to the front of `_LOCAL####`. This results in `os.path.splitext` not properly identifying the files to load. 

In order to not break any existing code that is formatted to work properly with `load_all`, we should still check if an extension is found using the existing method. If it is, we will utilize that, otherwise, we can use the optional `file_format` parameter to identify the file type and set the extension.

**Comment by Ethan-DeBandi99** [comment]
@bmcdonald3 - is there any specific reason you see to not add the file_format here if we continue to allow for the extensions to be checked as well?

**Comment by bmcdonald3** [comment]
No, I just overlooked this case I think, should be good to add the `file_format` keyword here as well.

**Comment by Ethan-DeBandi99** [comment]
10-4. I forgot about this too until adding a load feature for the dataframe stuff. I just wanted to make sure before updating that I wasn't missing something. Thanks!

**Comment by Ethan-DeBandi99** [comment]
I noticed that this also needs to be done for the `load` functions. I will update these as well.

**Comment by Ethan-DeBandi99** [comment]
Testing for these changes requires the same changes made to `ak.save_all()` and `ak.Categorical.save()` in #1303. Once PR #1309 is merged, I will push the PR for these changes.
